src/uploadAsset.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2021-2025 Littleton Robotics
+// http://github.com/Mechanical-Advantage
+//
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file
+// at the root directory of this project.
+
+window.addEventListener("message", (event) => {
+  const ASSET_INPUT = document.getElementById("asset") as HTMLInputElement;
+  const EXIT_BUTTON = document.getElementById("exit") as HTMLInputElement;
+  const CONFIRM_BUTTON = document.getElementById("confirm") as HTMLInputElement;
+  const PROGRESS_TEXT = document.getElementById("progress") as HTMLTableCellElement;
+
+  let messagePort = event.ports[0];
+  messagePort.onmessage = (event) => {
+    // Update button focus
+    if (typeof event.data === "object" && "isFocused" in event.data) {
+      Array.from(document.getElementsByTagName("button")).forEach((button) => {
+        if (event.data.isFocused) {
+          button.classList.remove("blurred");
+        } else {
+          button.classList.add("blurred");
+        }
+      });
+      return;
+    }
+  };
+
+  ASSET_INPUT.addEventListener("change", (event) => {
+    const files = ASSET_INPUT.files ?? [];
+    for (const file of files) {
+      uploadFile(file);
+    }
+  });
```

Comment:
Rather than uploading the assets as soon as the file is selected, I would recommend waiting until they push the "confirm" button. The typical UX for a file input like this is that the value can be freely changed before the upload starts.